chi/server.py

```diff
@@ -427,20 +429,43 @@ def show_server_by_name(name) -> NovaServer:
     return show_server(server_id)
 
 
-def associate_floating_ip(server_name, floating_ip_address=None):
-    server_id = get_server_id(server_name)
 
-    if floating_ip_address:
-        fip = get_floating_ip(floating_ip_address)
-    else:
+def associate_floating_ip(server_name, floating_ip_address=None):
+    #TODO: This function should add fip to a port, not a server
+    server = get_server(server_name)
+    
+    if floating_ip_address == None:
         fip = get_free_floating_ip()
-
+    else:
+        fip = get_floating_ip(floating_ip_address)
+  
+    if fip == None:
+        return
     ip = fip['floating_ip_address']
-    connection().compute.add_floating_ip_to_server(server_id, ip)
```

Comment:
Can you point to a specific error you got here? This is a good function: https://github.com/openstack/openstacksdk/blob/f4dd6fe5fd21fb866b43330ecbbafee5cf553ded/openstack/compute/v2/_proxy.py#L889-L902